File: satisfaction-eval/backend/app/schemas/domain.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class KpiCardData(BaseModel):
    title: str
    value: float | str
    change: float
    changeLabel: str
    icon: str
    color: str


class ModelMetrics(BaseModel):
    modelName: str
    accuracy: float
    recall: float
    f1: float
    auc: float
    psi: float
    updateTime: str


class RiskUser(BaseModel):
    userId: str
    region: str
    riskScore: float
    riskLevel: str
    topCauses: list[dict]
    lastEvent: str


class SurveyUser(BaseModel):
    userId: str
    region: str
    score: int
    level: str
    lastSurveyDate: str
    hitStatus: str


class ComplaintAlert(BaseModel):
    id: str
    region: str
    alertType: str
    riskLevel: str
    predictedCount: int
    actualCount: Optional[int] = None
    trend: str
    alertTime: str
    windowDays: int


class WorkOrder(BaseModel):
    id: str
    title: str
    type: str
    priority: str
    status: str
    assignee: str
    region: str
    createTime: str
    deadline: str
    relatedUserId: Optional[str] = None


class CaseItem(BaseModel):
    id: str
    title: str
    category: str
    tags: list[str]
    problem: str
    solution: str
    effect: str
    region: str
    createTime: str


class DataSource(BaseModel):
    code: str
    name: str
    type: str
    syncMode: str
    syncFreq: str
    status: str
    lastSync: str
    successRate: float


class FeatureDefinition(BaseModel):
    code: str
    name: str
    layer: str
    updateFreq: str
    version: str
    importance: Optional[float] = None


class ModelVersion(BaseModel):
    name: str
    version: str
    type: str
    algorithm: str
    status: str
    metrics: ModelMetrics
    deployTime: str

File: satisfaction-eval/src/types/domain.ts
export type Role = 'admin' | 'analyst' | 'operator' | 'auditor';

export type RiskLevel = 'high' | 'medium' | 'low';

export type TaskStatus = 'pending' | 'assigned' | 'in_progress' | 'resolved' | 'verified' | 'closed';

export type RegionLevel = 'province' | 'city' | 'county' | 'grid' | 'cell';

export interface Region {
  code: string;
  name: string;
  level: RegionLevel;
  parentCode?: string;
}

export interface KpiCardData {
  title: string;
  value: number | string;
  change: number;
  changeLabel: string;
  icon: string;
  color: 'blue' | 'green' | 'orange' | 'red' | 'purple';
}

export interface ModelMetrics {
  modelName: string;
  accuracy: number;
  recall: number;
  f1: number;
  auc: number;
  psi: number;
  updateTime: string;
}

export interface RiskUser {
  userId: string;
  region: string;
  riskScore: number;
  riskLevel: RiskLevel;
  topCauses: { code: string; name: string; contribution: number }[];
  lastEvent: string;
}

export interface SurveyUser {
  userId: string;
  region: string;
  score: number;
  level: string;
  lastSurveyDate: string;
  hitStatus: 'hit' | 'miss' | 'pending';
}

export interface ComplaintAlert {
  id: string;
  region: string;
  alertType: string;
  riskLevel: RiskLevel;
  predictedCount: number;
  actualCount?: number;
  trend: 'up' | 'down' | 'stable';
  alertTime: string;
  windowDays: number;
}

export interface WorkOrder {
  id: string;
  title: string;
  type: string;
  priority: RiskLevel;
  status: TaskStatus;
  assignee: string;
  region: string;
  createTime: string;
  deadline: string;
  relatedUserId?: string;
}

export interface CaseItem {
  id: string;
  title: string;
  category: string;
  tags: string[];
  problem: string;
  solution: string;
  effect: string;
  region: string;
  createTime: string;
}

export interface DataSource {
  code: string;
  name: string;
  type: string;
  syncMode: 'api' | 'ftp';
  syncFreq: string;
  status: 'active' | 'inactive' | 'error';
  lastSync: string;
  successRate: number;
}

export interface FeatureDefinition {
  code: string;
  name: string;
  layer: 'basic' | 'derived' | 'scenario';
  updateFreq: string;
  version: string;
  importance?: number;
}

export interface ModelVersion {
  name: string;
  version: string;
  type: 'dissatisfaction' | 'survey' | 'complaint_cls' | 'complaint_ts';
  algorithm: string;
  status: 'draft' | 'candidate' | 'prod' | 'rollback';
  metrics: ModelMetrics;
  deployTime: string;
}
